### ISSUE TITLE
Error running with Gradle 4.1 (and possibly Android's gradle plugin 3.0.0-beta7)

### ISSUE THREAD

**Issue by erawhctim** [radarsh/gradle-test-logger-plugin#14]
Adding the plugin to my android project throws the following error:

`Cannot cast object 'task ':app:test'' with class 'org.gradle.api.DefaultTask_Decorated' to class 'org.gradle.api.tasks.testing.Test'.`

Any ideas?

**Comment by radarsh** [comment]
Interesting. This seems to be related to #10.

**Comment by erawhctim** [comment]
I would be inclined to agree, though I didn't dig in to anything internal to Gradle or the android gradle plugin to confirm. 

Happy to provide any additional stacktrace/debug info if you need it 👍

**Comment by radarsh** [comment]
@erawhctim could you provide a very simple, stripped down sample project which demonstrates this issue? I have never done Android development so it would probably take me far longer to reproduce this than someone who is well into it.

**Comment by erawhctim** [comment]
Sure, [here you go](https://github.com/erawhctim/GradleLoggerAndroidTest)

You can see the plugin applied [in the build.gradle file here](https://github.com/erawhctim/GradleLoggerAndroidTest/blob/master/app/build.gradle#L4), running `./gradlew test` from the project root will show the gradle error. Comment that line out and run again and you'll see a success message. 

There's a [single unit test here](https://github.com/erawhctim/GradleLoggerAndroidTest/blob/master/app/src/test/java/com/gradlelogger/test/ExampleUnitTest.kt) which you can modify to fail on purpose if you need to test. 

Gradle version is set [here](https://github.com/erawhctim/GradleLoggerAndroidTest/blob/master/gradle/wrapper/gradle-wrapper.properties#L6) - that should be enough to get you started?

**Comment by radarsh** [comment]
This is brilliant. I will try a potential fix I have and let you know. Thanks for the help.

**Comment by radarsh** [comment]
Just confirmed that #10 has fixed this too. I got the below output when I used the development version of the plugin.

```bash
✔ 23:29 /d/workspace/GradleLoggerAndroidTest [master|✚ 1] $ ANDROID_HOME=/d/Android/User/Sdk ./gradlew clean test --stacktrace
:clean
:app:clean
:app:preBuild UP-TO-DATE
:app:preDebugBuild
:app:compileDebugAidl
:app:compileDebugRenderscript
:app:checkDebugManifest
:app:generateDebugBuildConfig
:app:generateDebugResValues
:app:generateDebugResources
:app:mergeDebugResources
:app:createDebugCompatibleScreenManifests
:app:processDebugManifest
:app:splitsDiscoveryTaskDebug
:app:processDebugResources
:app:compileDebugKotlin
:app:generateDebugSources
:app:javaPreCompileDebug
:app:compileDebugJavaWithJavac
:app:compileDebugUnitTestKotlin
Using Kotlin incremental compilation
:app:preDebugUnitTestBuild UP-TO-DATE
:app:javaPreCompileDebugUnitTest
:app:compileDebugUnitTestJavaWithJavac NO-SOURCE
:app:mockableAndroidJar
:app:processDebugJavaRes NO-SOURCE
:app:processDebugUnitTestJavaRes NO-SOURCE
:app:testDebugUnitTest

com.gradlelogger.test.ExampleUnitTest

  Test addition_isCorrect PASSED

:app:preReleaseBuild
:app:compileReleaseAidl
:app:compileReleaseRenderscript
:app:checkReleaseManifest
:app:generateReleaseBuildConfig
:app:generateReleaseResValues
:app:generateReleaseResources
:app:mergeReleaseResources
:app:createReleaseCompatibleScreenManifests
:app:processReleaseManifest
:app:splitsDiscoveryTaskRelease
:app:processReleaseResources
:app:compileReleaseKotlin
:app:generateReleaseSources
:app:javaPreCompileRelease
:app:compileReleaseJavaWithJavac
:app:compileReleaseUnitTestKotlin
Using Kotlin incremental compilation
:app:preReleaseUnitTestBuild UP-TO-DATE
:app:javaPreCompileReleaseUnitTest
:app:compileReleaseUnitTestJavaWithJavac NO-SOURCE
:app:processReleaseJavaRes NO-SOURCE
:app:processReleaseUnitTestJavaRes NO-SOURCE
:app:testReleaseUnitTest

com.gradlelogger.test.ExampleUnitTest

  Test addition_isCorrect PASSED

:app:test

BUILD SUCCESSFUL in 6s
37 actionable tasks: 37 executed
```
I have a few more issues to fix and then hopefully `1.0.1` will be out soon.

**Comment by erawhctim** [comment]
Awesome, thanks for the help and the awesome plugin!

**Comment by radarsh** [comment]
Sorry but currently the only way to do this is to build it locally. You can change the version in gradle.properties to something like `1.0.0.beta` and run `./gradlew clean publishToMavenLocal`

After it’s published you have to change your Android project to use this version. Also add `mavenLocal()` to the list of repositories inside the `buildscript` DSL.

**Comment by erawhctim** [comment]
Okay great, thanks for the info. And this fix will be part of v1.0.1 right?

**Comment by radarsh** [comment]
@erawhctim I have released `v1.0.1` of the plugin which has fix for this issue and a [number of others](https://github.com/radarsh/gradle-test-logger-plugin/releases/tag/v1.0.1).
